packages/tables/backend/sirius-components-collaborative-tables/src/main/java/org/eclipse/sirius/components/collaborative/tables/handlers/EditCellEventHandler.java

```diff
@@ -92,18 +90,24 @@ public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDesc
         String message = this.messageService.invalidInput(tableInput.getClass().getSimpleName(), EditCheckboxCellInput.class.getSimpleName());
         IPayload payload = new ErrorPayload(tableInput.id(), message);
 
-        if (tableInput instanceof EditCheckboxCellInput editCheckboxCellInput) {
-            var optCell = this.tableQueryService.findCellById(table, UUID.fromString(editCheckboxCellInput.cellId()));
+        if (tableInput instanceof IEditCellInput editCellInput) {
+            var optCell = this.tableQueryService.findCellById(table, UUID.fromString(editCellInput.cellId()));
 
             if (optCell.isPresent()) {
                 ICell cell = optCell.get();
                 var optLine = this.tableQueryService.findLineByCellId(table, cell.getId());
                 var optCol = this.tableQueryService.findColumnById(table, cell.getColumnId());
 
                 if (optLine.isPresent() && optCol.isPresent()) {
-                    this.invokeEditCell(cell, optLine.get(), optCol.get(), editingContext, tableDescription, editCheckboxCellInput.newValue());
-                    changeDescription = new ChangeDescription(ChangeKind.SEMANTIC_CHANGE, editCheckboxCellInput.representationId(), editCheckboxCellInput);
-                    payload = this.getPayload(editCheckboxCellInput.id());
+
+                    var status = this.editCellExecutors.stream()
+                            .filter(executor -> executor.canExecute(tableDescription, editCellInput))
+                            .findFirst()
+                            .map(executor -> executor.execute(editingContext, tableDescription, cell, optLine.get(), optCol.get(), editCellInput))
+                            .orElseGet(() -> new Failure(this.messageService.noEditCellExecutor()));
```

Comment:
Just leverage an existing error message or create something more generic because tons of pieces of code can have the same issue.